backend/enterprise/service/license.go

```diff
@@ -128,22 +128,28 @@ func (s *LicenseService) GetEffectivePlan() api.PlanType {
 
 // GetPlanLimitValue gets the limit value for the plan.
 func (s *LicenseService) GetPlanLimitValue(ctx context.Context, name enterprise.PlanLimit) int64 {
-	v, ok := enterprise.PlanLimitValues[name]
-	if !ok {
-		return 0
-	}
-
 	subscription := s.LoadSubscription(ctx)
-
-	limit := v[subscription.Plan]
-	if subscription.Trialing {
-		limit = v[api.FREE]
+	if subscription == nil || subscription.Plan == api.FREE {
+		// Free plan uses the maximum value from plan.yaml.
+		v, ok := enterprise.PlanLimitValues[name]
+		if !ok {
+			return 0
+		}
+		return v[api.FREE]
 	}
 
-	if limit == -1 {
+	switch name {
+	case enterprise.PlanLimitMaximumEnvironment:
 		return math.MaxInt64
+	case enterprise.PlanLimitMaximumUser:
+		if subscription.Seat == 0 {
+			return math.MaxInt64
+		}
+		return int64(subscription.Seat)
+	case enterprise.PlanLimitMaximumInstance:
```

Comment:
The instance count in the license is the instance license count, and the limit here is the maximum instance count (how many instances can users add), so I'm not sure about the changes here.

Reply:
I think the limit here is correct, it's a limit for total instance count. And the limit for active instance (instance with license) should at https://github.com/bytebase/bytebase/blob/fee98798e8f36878594af1d45fafc70238010988/backend/api/v1/instance_service.go#L186

Reply:
LGTM, but if the total instance count == instance license count, then what's the meaning of the instance license?🤣
I remind we add this by design, to let users add instances easily (even as many as possible), but guard some features via the instance license.

Reply:
We may rebrand the instance license to license which includes all license features (features, seats, instances). We don't guard the # of instances to be added, but guard the active instances under the license.